api_server/src/request/mod.rs

```diff
@@ -322,16 +322,14 @@ mod tests {
             StartMicrovmError::KernelLoader(kernel::loader::Error::BigEndianElfOnLittle),
         );
         check_error_response(vmm_resp, StatusCode::BadRequest);
+        let vmm_resp =
+            VmmActionError::StartMicrovm(ErrorKind::Internal, StartMicrovmError::CommandLineCopy);
+        check_error_response(vmm_resp, StatusCode::InternalServerError);
         let vmm_resp = VmmActionError::StartMicrovm(
-            ErrorKind::User,
-            StartMicrovmError::LoadCommandline(kernel::loader::Error::CommandLineCopy),
-        );
-        check_error_response(vmm_resp, StatusCode::BadRequest);
-        let vmm_resp = VmmActionError::StartMicrovm(
-            ErrorKind::User,
-            StartMicrovmError::LoadCommandline(kernel::loader::Error::CommandLineOverflow),
+            ErrorKind::Internal,
+            StartMicrovmError::CommandLineOverflow,
         );
-        check_error_response(vmm_resp, StatusCode::BadRequest);
+        check_error_response(vmm_resp, StatusCode::InternalServerError);
```

Comment:
It's rather unfortunate that this ends up being a `InternalServerError`. I believe ComandLineOverflow should be `BadRequest` because it is generated by the user memory & kernel configuration. Any chance you can leave it as it was before?

Reply:
Yes, good catch.

Reply:
done